writeaheadlog_test.go

```diff
@@ -179,6 +147,63 @@ func TestCommitFailed(t *testing.T) {
 	}
 }
 
+// TestWALInconsistentSync verifies that the WAL handles the case where the
+// file randomly fails writes. When the file failes a write, it returns an
+// error on sync.
+func TestWALInconsistentSync(t *testing.T) {
+	var wt *walTester
+	err := build.Retry(50, time.Millisecond*100, func() error {
+		tester, err := newWALTester(t.Name(), faultyDiskDependency{})
+		if err != nil {
+			return err
+		}
+		wt = tester
+		return nil
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer wt.Close()
+
+	// for 20 seconds, create large transactions and commit them to the wal, then
+	// reopen the WAL. The WAL should stay consistent.
+	for start := time.Now(); time.Since(start) < time.Second*10; time.Sleep(time.Millisecond * 100) {
```

Comment:
maybe clearer:
```go
ticker := time.NewTicker(100 * time.Millisecond)
time.AfterFunc(10 * time.Second, ticker.Stop)
for range ticker.C {
    // ...
}
```
or maybe that's too fancy :P

Reply:
ah damn the docstring says 20 seconds but the actual constant is 10 lol. I'll fix